### ISSUE TITLE
Auto-materialize history time doesn't respect user timezone

### ISSUE THREAD

**Issue by EtienneT** [dagster-io/dagster#18590]
### Dagster version

1.5.10

### What's the issue?

The time listed in the Auto-materialize history tab of an asset does not use the user setting timezone.

![image](https://github.com/dagster-io/dagster/assets/265924/b452a521-0dfa-4cd5-a313-ce0475e4effd)

### What did you expect to happen?

It should be adjusted according to the user settings timezone.

### Deployment type

None

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by johannkm** [comment]
@EtienneT can you confirm that the other timestamps in the app behave like you expect?

Where are you running dagster? What storage do you have set up?

**Comment by EtienneT** [comment]
I confirm that the other timestamps in the app work as expected and show time in the right timezone.

Running dagster locally with postgres as the storage.

**Comment by johannkm** [comment]
I suspect this is an issue with Postgres time zones. Could you confirm that
your Postgres is running on EST? You could also check if this is resolved
by switching Postgres to utc

**Comment by EtienneT** [comment]
If I run `show timezone` in postgres, I get "America/New_York".  So I set it to UTC by running `SET TIME ZONE 'UTC';`.  Then restarted dagster.  Same problem, I just tested a new auto-materialization and the time still shows 5h in the past.

**Comment by EtienneT** [comment]
It is kind of weird though that all the other dates were displayed correctly.  I guess they were also stored in postgres too, so I am wondering why they were different than this one.

Thanks for the quick investigation.

**Comment by johannkm** [comment]
Yeah we use a mix of the postgres datetime time which depends on our utc assumption and timestamps which don't. I expect you'd also see the wrong times in the 'created' column of the runs table.

Ideally we'd remove this requirement but so far it's been alright to require UTC

**Comment by EtienneT** [comment]
Makes 100% sense to require UTC everywhere, but it should be clear in the docs to setup postgres just to avoid confusion.

Thanks!
